crates/compiler/gen_llvm/src/llvm/lowlevel.rs

```diff
@@ -2876,3 +3005,25 @@ fn load_symbol_and_lambda_set<'a, 'ctx>(
         other => panic!("Not a lambda set: {other:?}, {ptr:?}"),
     }
 }
+use inkwell::types::BasicTypeEnum;
+///Change from a basic LLVM value to a Roc `Result`
+fn result_with<'ctx>(
+    env: &Env<'_, 'ctx, '_>,
+    good_type: BasicTypeEnum,
+    good_value: impl BasicValue<'ctx>,
+    bad_type: BasicTypeEnum,
+    bad_value: impl BasicValue<'ctx>,
+) -> StructValue<'ctx> {
+    let struct_type = env.context.struct_type(&[good_type, bad_type], false);
+    let return_value = struct_type.const_zero();
+    let return_value = env
+        .builder
+        .build_insert_value(return_value, good_value, 0, "good_value")
+        .unwrap();
+    let return_value = env
+        .builder
+        .build_insert_value(return_value, bad_value, 1, "is_bad")
+        .unwrap();
+
+    return_value.into_struct_value()
+}
```

Comment:
Can these unwraps be reasonably avoided so we don't crash the compiler with a bad error message?